src/NServiceBus.RabbitMQ.AcceptanceTests/ConfigureEndpointRabbitMQTransport.cs

```diff
@@ -24,34 +24,57 @@ class ConfigureEndpointRabbitMQTransport : IConfigureEndpointTestExecution
 
     public Task Configure(string endpointName, EndpointConfiguration configuration, RunSettings settings)
     {
-        connectionStringBuilder = new DbConnectionStringBuilder
+        var connectionString = Environment.GetEnvironmentVariable("RabbitMQTransport.ConnectionString");
+
+        if (string.IsNullOrEmpty(connectionString))
         {
-            ConnectionString = Environment.GetEnvironmentVariable("RabbitMQTransport.ConnectionString")
-        };
+            throw new Exception("The 'RabbitMQTransport.ConnectionString' environment variable is not set.");
+        }
 
-        ApplyDefault(connectionStringBuilder, "username", "guest");
-        ApplyDefault(connectionStringBuilder, "password", "guest");
-        ApplyDefault(connectionStringBuilder, "virtualhost", "nsb-rabbitmq-test");
-        ApplyDefault(connectionStringBuilder, "host", "localhost");
+        connectionStringBuilder = new DbConnectionStringBuilder { ConnectionString = connectionString };
 
         configuration.UseTransport<RabbitMQTransport>().ConnectionString(connectionStringBuilder.ConnectionString);
 
         return TaskEx.CompletedTask;
     }
 
-    static void ApplyDefault(DbConnectionStringBuilder builder, string key, string value)
+    public Task Cleanup() => PurgeQueues();
+
+    async Task PurgeQueues()
     {
-        if (!builder.ContainsKey(key))
+        if (connectionStringBuilder == null)
         {
-            builder.Add(key, value);
+            return;
         }
-    }
 
-    public Task Cleanup() => PurgeQueues();
+        var connectionFactory = new ConnectionFactory
+        {
+            AutomaticRecoveryEnabled = true
+        };
 
-    async Task PurgeQueues()
-    {
-        var connectionFactory = CreateConnectionFactory();
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("password", out value))
+        {
+            connectionFactory.Password = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("virtualhost", out value))
+        {
+            connectionFactory.VirtualHost = value?.ToString();
+        }
+
+        if (connectionStringBuilder.TryGetValue("host", out value))
+        {
+            connectionFactory.HostName = value?.ToString();
+        }
+
+        connectionFactory.ClientProperties["purpose"] = "Test Queue Purger";
```

Comment:
I don't see an obvious way of doing that in line 81.

I can see that `CreateConnection()` has an overload which takes a `string clientProvidedName` so perhaps that's what you mean, but there's nothing in the API which indicates that that is the same as `ClientProperties["purpose"]`.

BTW - this is creeping the scope for this PR, since the current code looks like this:

```C#
var connectionFactory = new ConnectionFactory
{
    UserName = connectionStringBuilder["username"].ToString(),
    Password = connectionStringBuilder["password"].ToString(),
    VirtualHost = connectionStringBuilder["virtualhost"].ToString(),
    HostName = connectionStringBuilder["host"].ToString(),
    AutomaticRecoveryEnabled = true
};

connectionFactory.ClientProperties["purpose"] = "Test Queue Purger";
```